### PR TITLE
Bump go version to 1.20

### PR DESCRIPTION
**Pull request by waciumawanjohi**
[PR requirements]: https://github.com/vmware-tanzu/cartographer/blob/main/CONTRIBUTING.md#commit-message-and-pr-guidelines

## Changes proposed by this PR

Stop pinning go version in github action. See https://github.com/vmware-tanzu/cartographer/pull/1267

## Release Note

[imperative mood]: https://chris.beams.io/posts/git-commit/#imperative

## Cherry-pick branches
0.6, 0.5

## PR Checklist

Note: Please do not remove items. Mark items as done `[x]` or use ~strikethrough~ if you believe they are not relevant

- [x] Linked to a relevant issue. Eg: `Fixes #123` or `Updates #123`
- [x] Removed non-atomic or `wip` commits
- ~[ ] Filled in the [Release Note](#Release-Note) section above~
- [x] Added any relevant branches to cherry-pick
- ~[ ] Modified the docs to match changes~

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       - name: set up Go
         uses: actions/setup-go@v4
         with:
-          go-version: '^1.19.9'
+          go-version-file: 'go.mod'
```
Comment (squeedee) [inline .github/workflows/release.yaml:33]: This has caused issues in the past with surprise non-repro. That is, we cant seem to build/test something that was working and can't figure it out. That's why we used a pinned minor. Would rather we use `^1.20.6`

**Review by squeedee — APPROVED** [review]

File: .github/workflows/release.yaml (hunk 1 above)
Comment (squeedee) [reply .github/workflows/release.yaml:33]: I had misread the change. After discussion, agree this is a great solution for unpinning but tracing a minor version.

Comment (waciumawanjohi) [reply .github/workflows/release.yaml:33]: To document: we chatted, this change uses the minor written into `go.mod`. While we could achieve patch pinning and auto-updating with UpdateCLI, we're satisfied with the minor pinning that we achieve with `go-version-file`.

Comment (waciumawanjohi) [reply .github/workflows/release.yaml:33]: Jinx

**Pull request merged**